aws-ssm-parameter/src/main/java/com/amazonaws/ssm/parameter/CreateHandler.java

```diff
@@ -0,0 +1,102 @@
+package com.amazonaws.ssm.parameter;
+
+import com.amazonaws.AmazonServiceException;
+import org.apache.commons.lang3.RandomStringUtils;
+
+import software.amazon.awssdk.services.ssm.SsmClient;
+import software.amazon.awssdk.services.ssm.model.ParameterAlreadyExistsException;
+import software.amazon.awssdk.services.ssm.model.InternalServerErrorException;
+import software.amazon.awssdk.services.ssm.model.PutParameterRequest;
+import software.amazon.awssdk.services.ssm.model.PutParameterResponse;
+import software.amazon.cloudformation.exceptions.CfnAlreadyExistsException;
+import software.amazon.cloudformation.exceptions.CfnGeneralServiceException;
+import software.amazon.cloudformation.exceptions.CfnServiceInternalErrorException;
+import software.amazon.cloudformation.exceptions.CfnThrottlingException;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+
+import java.util.Map;
+import java.util.Random;
+
+public class CreateHandler extends BaseHandlerStd {
+    private static final String OPERATION = "PutParameter";
+    private static final String RETRY_MESSAGE = "Detected retryable error, retrying. Exception message: %s";
+    private Logger logger;
+
+    @Override
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final ProxyClient<SsmClient> proxyClient,
+            final Logger logger) {
+        this.logger = logger;
+        final ResourceModel model = request.getDesiredResourceState();
+
+        Map<String, String> consolidatedTagList = request.getDesiredResourceTags();
+        consolidatedTagList.putAll(request.getSystemTags());
+
+        // Set model primary ID if absent
+        if(model.getName() == null) {
+            model.setName(generateParameterName(
+                    request.getLogicalResourceIdentifier(),
+                    request.getClientRequestToken()
+            ));
+        }
+
+        return proxy.initiate("aws-ssm-parameter::resource-create", proxyClient, model, callbackContext)
```

Comment:
We should probably add the `secure-string` unsupported condition here and in update handler, just to be extra-safe.